File: code/stable/Projectile/Projectile_S_L_NoL_U_U_V_F/src/java/Projectile/InputFormat.java
package Projectile;

/** \file InputFormat.java
    \author Samuel J. Crawford, Brooks MacLachlan, and W. Spencer Smith
    \brief Provides the function for reading inputs
*/
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class InputFormat {
    
    /** \brief Reads input from a file with the given file name
        \param filename name of the input file
        \return array containing the following values:
        \return launch speed: the initial speed of the projectile when launched (m/s)
        \return launch angle: the angle between the launcher and a straight line from the launcher to the target (rad)
        \return target position: the distance from the launcher to the target (m)
    */
    public static Object[] get_input(String filename) throws FileNotFoundException {
        float v_launch;
        float theta;
        float p_target;
        
        Scanner infile;
        infile = new Scanner(new File(filename));
        infile.nextLine();
        v_launch = Float.parseFloat(infile.nextLine());
        infile.nextLine();
        theta = Float.parseFloat(infile.nextLine());
        infile.nextLine();
        p_target = Float.parseFloat(infile.nextLine());
        infile.close();
        
        Object[] outputs = new Object[3];
        outputs[0] = v_launch;
        outputs[1] = theta;
        outputs[2] = p_target;
        return outputs;
    }
}

File: code/stable/projectile/projectile_s_l_nol_u_u_v_f/src/swift/InputFormat.swift
/** InputFormat.swift
    Provides the function for reading inputs
    - Authors: Samuel J. Crawford, Brooks MacLachlan, and W. Spencer Smith
*/
import Foundation

extension String: Error {}

/** Reads input from a file with the given file name
    - Parameter filename: name of the input file
    - Returns: launch speed: the initial speed of the projectile when launched (m/s)
    - Returns: launch angle: the angle between the launcher and a straight line from the launcher to the target (rad)
    - Returns: target position: the distance from the launcher to the target (m)
*/
func get_input(_ filename: String) throws -> (Float, Float, Float) {
    var v_launch: Float
    var theta: Float
    var p_target: Float
    
    var infile: URL
    infile = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!.appendingPathComponent(filename)
    var goolContents: [[String]]
    do {
        goolContents = try String(contentsOf: infile).components(separatedBy: "\n").map({(l: String) -> [String] in l.components(separatedBy: " ")})
    } catch {
        throw "Error reading from file."
    }
    v_launch = Float(goolContents[1][0])!
    theta = Float(goolContents[2][0])!
    p_target = Float(goolContents[3][0])!
    
    return (v_launch, theta, p_target)
}
